File: Problem_001/test_solution.cpp
#include <iostream>
#include <cassert>
#include <vector>
#include "solution.cpp"

void run_tests()
{
    // Test 1
    assert(hasPairWithSum({10, 15, 3, 7}, 17) == true);

    // Test 2
    assert(hasPairWithSum({10, 15, 3, 7}, 20) == false);

    // Test 3
    assert(hasPairWithSum({}, 10) == false);

    // Test 4
    assert(hasPairWithSum({10}, 10) == false);

    // Test 5
    assert(hasPairWithSum({-5, 10, 2, 8}, 3) == true);

    // Test 6
    assert(hasPairWithSum({0, 10, 5, 2}, 10) == true);

    // Test 7
    assert(hasPairWithSum({5, 2, 3}, 10) == false);
    assert(hasPairWithSum({5, 2, 5}, 10) == true);

    std::cout << "All tests passed successfully!" << std::endl;
}

int main()
{
    run_tests();
    return 0;
}

File: Problem_014/test_solution.js
const once = require('./solution');

// Test cases
function runTests() {
    // Test 1: Basic functionality
    let fn1 = (a, b, c) => a + b + c;
    let onceFn1 = once(fn1);
    
    console.assert(
        onceFn1(1, 2, 3) === 6,
        'Test 1a Failed: First call should return 6'
    );
    
    console.assert(
        onceFn1(2, 3, 6) === undefined,
        'Test 1b Failed: Second call should return undefined'
    );

    // Test 2: Single parameter function
    let fn2 = (x) => x * 2;
    let onceFn2 = once(fn2);
    
    console.assert(
        onceFn2(5) === 10,
        'Test 2a Failed: First call should return 10'
    );
    
    console.assert(
        onceFn2(10) === undefined,
        'Test 2b Failed: Second call should return undefined'
    );

    // Test 3: No parameters function
    let fn3 = () => 'hello';
    let onceFn3 = once(fn3);
    
    console.assert(
        onceFn3() === 'hello',
        'Test 3a Failed: First call should return "hello"'
    );
    
    console.assert(
        onceFn3() === undefined,
        'Test 3b Failed: Second call should return undefined'
    );

    console.log('All tests passed!');
}

runTests();